src/main/java/org/fungover/thunder/PackageReader.java

```diff
@@ -18,21 +17,21 @@ public boolean isValidConnection(Socket socket) throws IOException {
         byte[] buffer = new byte[1024];
         InputStream inputStream = socket.getInputStream();
         OutputStream outputStream = socket.getOutputStream();
-        int bytesRead = inputStream.read(buffer);
 
         if (isDoubleConnectMessage(client, buffer)) {
```

Comment:
This code might have to be moved since its using buffer for something before we have read anything into the buffer?